### PR TITLE
Add missing links and examples for path modules and structs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @frewsxcv

### CODE REVIEW

**Comment by frewsxcv** [comment]
Looks great! Thanks!

@bors r+ rollup

File: src/libstd/path.rs [hunk 1]
```diff
@@ -10,20 +10,20 @@
 
 //! Cross-platform path manipulation.
 //!
-//! This module provides two types, `PathBuf` and `Path` (akin to `String` and
-//! `str`), for working with paths abstractly. These types are thin wrappers
-//! around `OsString` and `OsStr` respectively, meaning that they work directly
+//! This module provides two types, [`PathBuf`] and [`Path`] (akin to [`String`] and
```
Comment (sanxiyn) [inline src/libstd/path.rs]: `[Path] (akin to...)` seems to be interpreted as a Markdown link, causing Travis failure in linkcheck.

**Review by sanxiyn — CHANGES_REQUESTED** [review]

File: src/libstd/path.rs (hunk 1 above)
Comment (GuillaumeGomez) [inline src/libstd/path.rs]: Well found!

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/libstd/path.rs [hunk 2]
```diff
@@ -1912,6 +1963,15 @@ impl Path {
     /// This is an alias to [`fs::read_link`].
     ///
     /// [`fs::read_link`]: ../fs/fn.read_link.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::{Path, PathBuf};
```
Comment (frewsxcv) [inline src/libstd/path.rs]: Unused import: `PathBuf`

File: src/libstd/path.rs [hunk 3]
```diff
@@ -1927,6 +1987,19 @@ impl Path {
     /// [`io::Result`]: ../io/type.Result.html
     /// [`DirEntry`]: ../fs/struct.DirEntry.html
     /// [`fs::read_dir`]: ../fs/fn.read_dir.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::{Path, PathBuf};
```
Comment (frewsxcv) [inline src/libstd/path.rs]: Unused import: `PathBuf`

**Review by frewsxcv — CHANGES_REQUESTED** [review]

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/libstd/path.rs [hunk 4]
```diff
@@ -1902,6 +1944,15 @@ impl Path {
...
     ///
     /// [`fs::canonicalize`]: ../fs/fn.canonicalize.html
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::Path;
+    ///
+    /// let path = Path::new("/foo/test/../test/bar.rs");
+    /// assert_eq!(path.canonicalize().unwrap(), PathBuf::from("/foo/test/bar.rs"));
```
Comment (frewsxcv) [inline src/libstd/path.rs]: Missing import `PathBuf`

Comment (GuillaumeGomez) [inline src/libstd/path.rs]: Oh damn, we'll never end it haha.

File: src/libstd/path.rs (hunk 3 above)
Comment (GuillaumeGomez) [inline src/libstd/path.rs]: It is used.

Comment (GuillaumeGomez) [inline src/libstd/path.rs]: (certainly github bug actually)

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by sanxiyn — APPROVED** [review]

**Pull request merged**